src/index.js

```diff
@@ -183,20 +183,56 @@ export default function (babel) {
 
 
   // HELPER FUNCTIONS
+  function visitTailExpressions(path, allowLoops, visitor) {
+    path.resync();
 
-  function transformTerminalExpressionsIntoArrPush(path, arrId) {
-    path.resync(); // not sure if this is necessary... c/p from addImplicitReturns
+    let tailPaths = getTailExpressions(path.get("body"), allowLoops);
+    for (let tailPath of tailPaths) {
+      if (tailPath.isExpressionStatement()) {
+        visitor(tailPath);
+      } else if (tailPath.isFunctionDeclaration()) {
+        // Promote function declarations to exprs.
```

Comment:
The core reason this code exists is because you can't insert anything that's not an expr into the `CallExpresion` for array.push or the `AssignExpression` for obj[k]=v without failing Babel validation. So whatever we return from here it has to be an expression, and there isn't anyway around that.